crates/router/src/connector/nexinets/transformers.rs

```diff
@@ -163,29 +165,70 @@ pub struct ApplepayPaymentMethod {
     token_type: String,
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for NexinetsPaymentsRequest {
+#[derive(Debug, Serialize)]
+pub struct NexinetsRouterData<T> {
+    amount: i64,
+    router_data: T,
+}
+impl<T>
+    TryFrom<(
+        &types::api::CurrencyUnit,
+        types::storage::enums::Currency,
+        i64,
+        T,
+    )> for NexinetsRouterData<T>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        let return_url = item.request.router_return_url.clone();
+    fn try_from(
+        (_currency_unit, _currency, amount, item): (
+            &types::api::CurrencyUnit,
+            types::storage::enums::Currency,
+            i64,
+            T,
+        ),
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount,
+            router_data: item,
+        })
+    }
+}
+
+impl
+    TryFrom<
+        &NexinetsRouterData<
+            &types::RouterData<
+                types::api::payments::Authorize,
+                PaymentsAuthorizeData,
+                PaymentsResponseData,
+            >,
+        >,
+    > for NexinetsPaymentsRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &NexinetsRouterData<
+            &types::RouterData<
+                types::api::payments::Authorize,
+                PaymentsAuthorizeData,
+                PaymentsResponseData,
+            >,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let return_url = item.router_data.request.router_return_url.clone();
         let nexinets_async = NexinetsAsyncDetails {
             success_url: return_url.clone(),
             cancel_url: return_url.clone(),
             failure_url: return_url,
         };
-        let (payment, product) = get_payment_details_and_product(item)?;
-        let merchant_order_id = match item.payment_method {
-            // Merchant order id is sent only in case of card payment
-            enums::PaymentMethod::Card => Some(item.connector_request_reference_id.clone()),
-            _ => None,
-        };
+        let (payment, product) = get_payment_details_and_product(&item.router_data)?;
         Ok(Self {
-            initial_amount: item.request.amount,
-            currency: item.request.currency,
+            initial_amount: item.router_data.request.amount,
```

Comment:
Here you have to directly use `item.amount` as we have already made the transformation

```suggestion
            initial_amount: item.amount,
```

And similarly you have to implement for `Capture`, `Void` and `Refund` Requests

Reply:
You can get it by searching for `amount` keyword in the file